src/main/java/sayo/Sayo.java

```diff
@@ -0,0 +1,176 @@
+                        String description = input.substring(9, byIndex).trim();
+                        String by = input.substring(byIndex + 4).trim();
+                        Deadline newDeadline = new Deadline(description, by);
+                        items.add(newDeadline); // Add directly to the ArrayList
+                        System.out.println("Got it. I've added this task:");
+                        System.out.println("  " + newDeadline);
+                        System.out.println("Now you have " + items.size() + " tasks in the list.");
+                    }
+                    saveTasksToFile();
+                } else if (input.startsWith("event")) {
+                    int fromIndex = input.indexOf("/from");
+                    int toIndex = input.indexOf("/to");
+
+                    // Check if the format of the input is correct
+                    if (fromIndex == -1 || toIndex == -1 || fromIndex + 6 >= input.length()
+                            || toIndex + 4 >= input.length()) {
+                        System.out.println(
+                                "Please enter the event in the format: event <event description> /from <start time> /to <end time>");
+                    } else {
+                        String description = input.substring(6, fromIndex).trim();
+                        String start = input.substring(fromIndex + 6, toIndex).trim();
+                        String end = input.substring(toIndex + 4).trim();
+
+                        // Create a new Event object and add it to the list
+                        Event newEvent = new Event(description, start, end);
+                        items.add(newEvent);
+
+                        System.out.println("Got it. I've added this task:");
+                        System.out.println("  " + newEvent);
+                        System.out.println("Now you have " + items.size() + " tasks in the list.");
+                    }
+                    saveTasksToFile();
+                } else if (input.startsWith("delete")) {
+                    try {
+                        int index = Integer.parseInt(input.substring(7).trim()) - 1; 
+                        if (index >= 0 && index < items.size()) {
+                            Task removedTask = items.remove(index); // Remove the task and retrieve it for printing
+                            System.out.println("Noted. I've removed this task: ");
+                            System.out.println(removedTask);
+                            System.out.println("Now you have " + items.size() + " tasks in the list.");
+                            saveTasksToFile(); 
+                        } else {
+                            System.out.println("Invalid task number. Please enter a valid task number to delete.");
+                        }
+                    } catch (NumberFormatException e) {
+                        System.out.println("Please enter a valid number after 'delete'.");
+                    }
```

Comment:
Try to avoid deep nesting. Perhaps you could use abstraction to make it more readable.